common/nalreader_unittest.cpp

```diff
@@ -53,20 +53,20 @@ const std::array<uint8_t, 39> g_data = {
     0x12
 };
 
-const std::array<int32_t, 5> g_nsizes = {9, 4, 6, 12, 7};
+const std::array<int32_t, 5> g_nsizes = { 6, 1, 3, 9, 4 };
 
 NALREADER_TEST(ReadAsUnits) {
     const uint8_t* nal;
-    int32_t offset(1), size;
+    int32_t offset(0), size;
 
-    NalReader reader(&g_data[0], g_data.size(), false);
+    NalReader reader(&g_data[0], g_data.size(), 0, false);
 
     for (unsigned n(0); n < g_nsizes.size(); ++n) {
         EXPECT_TRUE(reader.read(nal, size));
         EXPECT_EQ(size, g_nsizes[n]);
         for (int32_t i(0); i < size; ++i)
-            EXPECT_EQ(nal[i], g_data[i+offset]);
-        offset += size;
+            EXPECT_EQ(nal[i], g_data[i + offset + 4]);
```

Comment:
hi @xuguangxin , @uartie, as you see, 4 is const of 1 junk byte, and 3 start code bytes.